### PR TITLE
Companion Inputs and Mixes add Clear/Set/Invert all Flightmodes #3305

### PR DESCRIPTION
**Pull request by elecpower**
This is the Companion part of Issue #3305 enhancement request.
I will look at the Radio part of the request but may need some advice/help.
Removed a block of what I can only determine to be test code from Inputs - advise me if a mistake.

### CODE REVIEW

File: companion/src/modeledit/expodialog.cpp [hunk 1]
```diff
@@ -140,71 +144,84 @@ void ExpoDialog::updateScale()
...
+  contextMenu.addAction(tr("Clear All"), this, SLOT(fmClearAll()));
+  contextMenu.addAction(tr("Set All"), this, SLOT(fmSetAll()));
+  contextMenu.addAction(tr("Invert All"), this, SLOT(fmInvertAll()));
+  contextMenu.exec(globalPos);
+}
+
+void ExpoDialog::fmClearAll()
+{
+  lock = true;
+  QCheckBox * cb_fp[] = {ui->cb_FP0,ui->cb_FP1,ui->cb_FP2,ui->cb_FP3,ui->cb_FP4,ui->cb_FP5,ui->cb_FP6,ui->cb_FP7,ui->cb_FP8 };
```
Comment (bsongis) [inline companion/src/modeledit/expodialog.cpp]: perhaps you could make a method which would return this array? in order to avoid duplicated code

**Comment by elecpower** [comment]
Took onboard @bsongis feedback but then struck compiler limitation on Ubuntu 14.04 gcc 4.8.4 im coding and added a version check to provide a temporary workaround.
Used flight modes max macro to future proof.

File: companion/src/modeledit/mixerdialog.cpp [hunk 2]
```diff
@@ -15,7 +15,21 @@ MixerDialog::MixerDialog(QWidget *parent, ModelData & model, MixData *mixdata, G
     ui->setupUi(this);
     QRegExp rx(CHAR_FOR_NAMES_REGEX);
     QLabel * lb_fp[] = {ui->lb_FP0,ui->lb_FP1,ui->lb_FP2,ui->lb_FP3,ui->lb_FP4,ui->lb_FP5,ui->lb_FP6,ui->lb_FP7,ui->lb_FP8 };
-    QCheckBox * cb_fp[] = {ui->cb_FP0,ui->cb_FP1,ui->cb_FP2,ui->cb_FP3,ui->cb_FP4,ui->cb_FP5,ui->cb_FP6,ui->cb_FP7,ui->cb_FP8 };
+    
+#if __cplusplus >= 201103L
+    //  extended initializer lists only available with -std=c++11 or -std=gnu++11
+    cb_fp[] = {ui->cb_FP0,ui->cb_FP1,ui->cb_FP2,ui->cb_FP3,ui->cb_FP4,ui->cb_FP5,ui->cb_FP6,ui->cb_FP7,ui->cb_FP8 };
```
Comment (elecpower) [inline companion/src/modeledit/mixerdialog.cpp]: Forced C++11 was what I thought had been implemented. I have a sneaking suspicion in a tired state I was compiling with make not cmake so will recompile tonight.

Comment (projectkk2glider) [inline companion/src/modeledit/mixerdialog.cpp]: We force C++11 in 2.2 only. No wonder it does not work in 2.1.

**Comment by bsongis** [comment]
Ok I didn't see it was for 'master' branch. Perhaps we can force C++1 in 2.1 there as well?

**Comment by bsongis** [comment]
@elecpower would you port this PR to 'next' which is currently our active branch? Thanks in advance

**Comment by elecpower** [comment]
@bsongis okay but will not be able to work on until next week

**Comment by elecpower** [comment]
Ported to next #4494. This one can be closed when you are ready

**Comment by elecpower** [comment]
Closed without merging into master as development moved to next

**Pull request closed without merging**